opentelekomcloud/resource_opentelekomcloud_rds_instance_v1.go

```diff
@@ -0,0 +1,618 @@
+		//if instance.Flavor.Id == flavorID {
+		//    return instance, "FLAVOR_UPDATED", nil
+		//}
+
+		return instance, instance.Status, nil
+	}
+}
+
+func resourceInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.RdsV1Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("Error Updating OpenTelekomCloud rds client: %s ", err)
+	}
+
+	log.Printf("[DEBUG] Updating instances %s", d.Id())
+	id := d.Id()
+
+	if d.HasChange("volume") {
+		var updateOpts instances.UpdateOps
+		volume := make(map[string]interface{})
+		volumeRaw := d.Get("volume").([]interface{})
+		log.Printf("[DEBUG] volumeRaw: %+v", volumeRaw)
+		if len(volumeRaw) == 1 {
+			volume["size"] = volumeRaw[0].(map[string]interface{})["size"].(int)
+		}
+		log.Printf("[DEBUG] volume: %+v", volume)
+		updateOpts.Volume = volume
+		_, err = instances.UpdateVolumeSize(client, updateOpts, id).Extract()
+		if err != nil {
+			return fmt.Errorf("Error updating instance volume from result: %s ", err)
+		}
+
+		stateConf := &resource.StateChangeConf{
+			Pending:    []string{"ACTIVE"},
+			Target:     []string{"UPDATED"},
+			Refresh:    InstanceStateUpdateRefreshFunc(client, id, updateOpts.Volume["size"].(int)),
+			Timeout:    d.Timeout(schema.TimeoutCreate),
+			Delay:      15 * time.Second,
+			MinTimeout: 3 * time.Second,
+		}
+
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			return fmt.Errorf(
+				"Error waiting for instance (%s) volume to be Updated: %s ",
+				id, err)
+		}
+		log.Printf("[DEBUG] Successfully updated instance %s volume: %+v", id, volume)
+	}
+
+	if d.HasChange("flavorref") {
+		var updateFlavorOpts instances.UpdateFlavorOps
+
+		log.Printf("[DEBUG] Update flavorref: %s", d.Get("flavorref").(string))
+
+		updateFlavorOpts.FlavorRef = d.Get("flavorref").(string)
+		_, err = instances.UpdateFlavorRef(client, updateFlavorOpts, id).Extract()
+		if err != nil {
+			return fmt.Errorf("Error updating instance Flavor from result: %s ", err)
+		}
+
+		stateConf := &resource.StateChangeConf{
+			Pending:    []string{"MODIFYING"},
+			Target:     []string{"ACTIVE"},
+			Refresh:    InstanceStateFlavorUpdateRefreshFunc(client, id, d.Get("flavorref").(string)),
+			Timeout:    d.Timeout(schema.TimeoutCreate),
+			Delay:      15 * time.Second,
+			MinTimeout: 3 * time.Second,
+		}
+
+		_, err = stateConf.WaitForState()
+		if err != nil {
+			return fmt.Errorf(
+				"Error waiting for instance (%s) flavor to be Updated: %s ",
+				id, err)
+		}
+		log.Printf("[DEBUG] Successfully updated instance %s flavor: %s", id, d.Get("flavorref").(string))
+	}
+
+	if d.HasChange("backupstrategy") {
+		var updatepolicyOpts instances.UpdatePolicyOps
+		backupstrategyRaw := d.Get("backupstrategy").([]interface{})
+		log.Printf("[DEBUG] backupstrategyRaw: %+v", backupstrategyRaw)
+		if len(backupstrategyRaw) == 1 {
+			updatepolicyOpts.StartTime = backupstrategyRaw[0].(map[string]interface{})["starttime"].(string)
```

Comment:
This is a potentially fragile way to do a type assertion (is it possible for the value to be `nil`)? May need to use the version that returns `ok` but I'll leave that up to you. As it stands this could potentially cause a panic.